.github/setup/action.yaml

```diff
@@ -0,0 +1,35 @@
+name: Install project dependencies
+
+inputs:
+  python-version:
+    required: true
+  os:
+    required: true
+
+runs:
+  using: composite
+  steps:
+
+    - name: Set up Python
+      uses: actions/setup-python@v4
+      with:
+        python-version: ${{ inputs.python-version }}
+
+    - name: Install Poetry
+      shell: bash
```

Comment:
don't ask me why it needs a shell and why bash also works on windows. Just mirrored a similar file in the pandas repository :)

Reply:
I don't think you need the shell here.  We aren't using it in `test.yaml`.  You only need the shell if you want to execute a unix command that is not available on Windows. 

Can you try deleting the references to the shell and see if it still works?

Reply:
I think that is a difference between workflow files and "composite actions". Fails without it.

Reply:
> I think that is a difference between workflow files and "composite actions". Fails without it.

seems to be